core/trino-main/src/main/java/io/trino/operator/scalar/SequenceFunction.java

```diff
@@ -120,6 +120,30 @@ private static Block fixedWidthSequence(long start, long stop, long step, FixedW
         return blockBuilder.build();
     }
 
+    private static int getLength(long start, long stop, long step)
+    {
+        if ((step > 0 && (start >= 0 || stop <= 0))
+                || (step < 0 && (start <= 0 || stop >= 0))) {
+            return checkMaxEntry((stop - start) / step + 1L);
```

Comment:
Actually, it could also overflow on addition. Please adjust the if-condition further to avoid it and use exact arithmetics

Reply:
We handled the step is 1 or -1 in advance, so here only need to make sure `stop - start` not overflow.